tests/services/test_devices.py

```diff
@@ -163,3 +164,13 @@ def test_get_page_when_org_not_found_raises_expected_error(
 
         assert "The organization with UID 'TestOrgUid' was not found." in str(err.value)
         assert err.value.org_uid == "TestOrgUid"
+
+    def test_upgrade_calls_upgrade_with_uri_and_params(
+        self, mock_connection, successful_response
+    ):
+        mock_connection.get.return_value = successful_response
+        service = DeviceService(mock_connection)
+        service.upgrade("DEVICE_ID")
+        expected_params = {"deviceGuid": "DEVICE_ID"}
+        uri = f"{UPGRADE_URI}"
```

Comment:
```suggestion
        uri = UPGRADE_URI
```

Reply:
or can just pass `UPGRADE_URI` to `assert_called_once_with`

Reply:
That makes sense, updated!